tests/tensors/test_ckks_vector.py

```diff
@@ -43,6 +43,98 @@ def test_negate_inplace(context, plain_vec):
     assert _almost_equal(decrypted_result, expected, 1), "Decryption of vector is incorrect"
 
 
+@pytest.mark.parametrize(
+    "plain_vec, power",
+    [
+        ([], 2),
+        ([0], 3),
+        ([0, 1, -1, 2, -2], 0),
+        ([1, -1, 2, -2], 1),
+        ([1, -1, 2, -2], 2),
+        ([1, -1, 2, -2], 3),
+        ([1, -2, 3, -4], 1),
+        ([1, -2, 3, -4], 2),
+        ([1, -2, 3, -4], 3),
```

Comment:
raising to the power of 4 should be also possible with the default parameters, we need a test-case for this to make sure the circuit is optimum. Same for the other tests

Reply:
Yeah, this should work with power of 4 as well.

Reply:
still missing a testcase of 4 here